### PR TITLE
Smoke test to ensure Ingress works

### PR DESCRIPTION
**Pull request by sablumiah**
WHAT
Setup a test that creates a temporary namespace, launches a service with an HTTP listener, a Service and an Ingress and test that you can hit the URL and get a 200 OK and then clean up namespace.

WHY
This will eventually be used as an automated test run on a schedule to test ingress on a given cluster

### CODE REVIEW

File: tests/01-ingress-http-traffic/ing-smoketest.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+kubectl create namespace ingress-smoketest
+
+sleep 3
+
+kubectl -n ingress-smoketest apply -f ./ingress-smoketest.yaml
+
+sleep 3
+
+output=$(curl -s -o /dev/null -w "%{http_code}" https://iingress-smoketest-app.apps.cloud-platform-test-1.k8s.integration.dsd.io)
```
Comment (kalbir) [inline tests/01-ingress-http-traffic/ing-smoketest.sh]: Is that a spelling mistake (`iingress`) - should it be

`https://ingress-smoketest-app.apps.cloud-platform-test-1.k8s.integration.dsd.io`

**Review by kalbir — CHANGES_REQUESTED** [review]
I wonder whether the output is verbose enough in a failure. I tested this locally and it failed because of the spelling mistake, this is the output that I got:

screen shot 2018-10-03 at 10 42 08

Maybe if we run `curl -v` in this case it would be better? Or is there a better way of getting some output here?

File: tests/01-ingress-http-traffic/ing-smoketest.sh (hunk 1 above)
Comment (sablumiah) [reply tests/01-ingress-http-traffic/ing-smoketest.sh]: Yes, that was the test scenario that was not changed back - fixed

**Comment by sablumiah** [comment]
@kalbir Added curl -v on failure to output whats on the page in addition to keeping NS on failure

**Comment by kalbir** [comment]
Do we need to rename the main test file `test.sh` (or add a `test.sh`) as per our conversation with @jojuolape and @alkar ?

**Comment by sablumiah** [comment]
i have renamed the test file to test.sh for now, we can change the test file name back and add a higher level test.sh if required depending on how we group all other tests together for automated testing

**Pull request merged**